### ISSUE TITLE
[bug] helloworld demo failed when "go mod tidy",  tried go 1.23.3 on both win10 and ubuntu.

### ISSUE THREAD

**Issue by shellohunter** [AllenDang/giu#922]
### What happend?

$ go mod tidy
```
go: finding module for package github.com/AllenDang/giu
go: found github.com/AllenDang/giu in github.com/AllenDang/giu v0.11.0
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/build
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/src
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/example/grep
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/example/timer
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/performance
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/include/boost
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/performance/config
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/example/snippets
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/c_compiler_checks
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/captures
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/collate_info
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/concepts
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/config_info
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/de_fuzz
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/named_subexpressions
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/object_cache
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/pathology
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/static_mutex
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/unicode
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/tools/generate
go: finding module for package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/regress
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/build: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/build
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/example/grep: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/example/grep
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/example/snippets: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/example/snippets
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/example/timer: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/example/timer
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/include/boost: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/include/boost
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/performance: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/performance
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/performance/config: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/performance/config
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/src: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/src
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/c_compiler_checks: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/c_compiler_checks
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/captures: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/captures
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/collate_info: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/collate_info
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/concepts: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/concepts
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/config_info: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/config_info
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/de_fuzz: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/de_fuzz
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/named_subexpressions: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/named_subexpressions
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/object_cache: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/object_cache
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/pathology: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/pathology
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/regress: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/regress
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/static_mutex: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/static_mutex
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/unicode: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/test/unicode
go: giu-test imports
        github.com/AllenDang/giu imports
        github.com/AllenDang/cimgui-go/imgui imports
        github.com/AllenDang/cimgui-go imports
        github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/tools/generate: module github.com/AllenDang/cimgui-go@latest found (v1.2.0), but does not contain package github.com/AllenDang/cimgui-go/cwrappers/ImGuiColorTextEdit/vendor/regex/tools/generate

```

### Code example

<details><summary><code>main.go</code></summary>

```golang
package main

import (
        "fmt"

        g "github.com/AllenDang/giu"
)

func onClickMe() {
        fmt.Println("Hello world!")
}

func onImSoCute() {
        fmt.Println("Im sooooooo cute!!")
}

func loop() {
        g.SingleWindow().Layout(
                g.Label("Hello world from giu"),
                g.Row(
                        g.Button("Click Me").OnClick(onClickMe),
                        g.Button("I'm so cute").OnClick(onImSoCute),
                ),
        )
}

func main() {
        wnd := g.NewMasterWindow("Hello world", 400, 200, g.MasterWindowFlagsNotResizable)
        wnd.Run(loop)
}

```

</details>

### To Reproduce

just trying the hello world demo.

tried: go-1.22.4, go-1.23.3, win10, linux, same error

### Version

master

### OS

win10, ubuntu

**Comment by gucio321** [comment]
This is a known bug causes by go itself. According to go'# issue this will be fixed in go 1.24 in February.

 See https://github.com/AllenDang/cimgui-go/issues/390 for details 

You can either not use go mod ridy or use v0.10.0 which should work

**Comment by shellohunter** [comment]
wow, thanks for your info.
that is the 1st time I came across a golang bug.
